doc/auth_design.md

```diff
@@ -0,0 +1,113 @@
+# Design: SQLFlow Authentication and Authorization
+
+## Terminology and Background
+
+For a quick explanation, the word "Authentication" means to identify the
+user recognized by the system is the actual real-world user. And "Authorization"
+means to grant privileges to the user to access some part of the system
+functionalities.
+
+SQLFlow works as a "bridge" between databases and
+Deep Learning/Machine Learning frameworks. In order to execute a job,
+SQLFlow need both permissions to access databases and submit jobs to
+systems to run distributed training jobs, like submitting jobs to Kubernetes
+to run a distributed tensorflow job.
+
+<img src="figures/auth1.png">
+
+In production environments, one SQLFlow server is designed to accept many clients'
```

Comment:
Maybe we can implement the Authentication and Authorization separated, for my brief idea:

![image](https://user-images.githubusercontent.com/1426912/59274626-4e183f00-8c8d-11e9-8dff-0db991e940fc.png)

1. Uses go the SQLFlow website such as `https://sqlflow.domain.com`, the auth-server would process the request and check the user.
2. If the user has not logged in, auth-server would redirect to the SSO URL with 302 redirect.
3. If the user has logged in, the SQLFlow auth-server (maybe another name) would launch the notebook if not exists with fill in the user token as environment vars.
4. notebook would call the SQLFlow server with Session struct (fill the user token)
5. SQLFlow server instance would auth MySQL/kubernetes and etc. with the user token.

Reply:
Updated the design with some modifications.